oshdb-filter/README.md

```diff
@@ -0,0 +1,108 @@
+
+Operators follow the following order of precedence: parentheses before `not`, before `and`, before `or`.
+
+### Special Characters & Whitespace
+
+When writing filters, tags without special characters can be supplied directly, without needing to quote them. Example: `amenity=drinking_water` or `name:it=*`. Allowed characters are: the letters `a-z` and `A-Z`, digits, underscore, dashes and colons.
+When filtering by tags with any other characters in their key or value, these strings need to be supplied as double-quoted strings, e.g. `name="Heidelberger Brückenaffe"` or `opening_hours="24/7"`. Escape sequences can be used to represent a literal double-quote character `\"`, while a literal backslash is written as `\\`.
+
+Whitespace such as spaces, tabs or newlines can be put freely between operators or parts of selectors (`name = *` is equivalent to `name=*`) to make a filter more readable.
+
+Examples
+--------
+
+Here's some useful examples for filtering some OSM features:
+
+| OSM Feature | filter | comment |
+|-------------|--------|---------|
+| forests/woods | `(landuse=forest or natural=wood) and geometry:polygon` | Using `geometry:polygon` will select closed ways as well as multipolygons (e.g. a forest with clearings). |
+| parks and park benches | `leisure=park and geometry:polygon or amenity=bench and (geometry:point or geometry:line)` | A filter can also fetch features of different geometry types: this returns parks (polygons) as well as park benches (points or lines). |
+| buildings | `building=* and building!=no and geometry:polygon` | This filter excludes the (rare) objects marked with `building=no`, which is a tag used to indicate that a feature might be expected to be a building (e.g. from an outdated aerial imagery source), but is in reality not one. |
+| highways | `type:way and highway in (motorway, motorway_link, trunk, trunk_link, primary, primary_link, secondary, secondary_link, tertiary, tertiary_link, unclassified, residential, living_street, pedestrian) or (highway=service and service=alley))` | The list of used tags depends on the exact definition of a "highway". In a different context, it may also incude less or even more highway tags (like `footway`, `cycleway`, `track`, `path`, all `highway=service`, etc.). |
+| residential roads missing a name (for quality assurance) | `type:way and highway=residential and name!=* and noname!=yes` | Note that some roads might be actually unnamed in reality. Such features can be marked as unnamed with the [`noname`](https://wiki.openstreetmap.org/wiki/Key:noname) tag in OSM. |
+| buildings with implausibly big footprints | `geometry:polygon and building=* and building!=no and area:(1E6..)` | The currently largest building by footprint area is a [car factory measuring about 887,800 m²](https://en.wikipedia.org/wiki/List_of_largest_buildings#Largest_footprint). |
+
```

Comment:
Here was a Reference to the javadoc. Do we need another reference here or is the general reference sufficient?

Reply:
I guess it should be fine like this. the reference was just here from when the filter was it's own repo.